paimon-core/src/main/java/org/apache/paimon/deletionvectors/DeletionVector.java

```diff
@@ -98,40 +100,55 @@ default boolean checkedDelete(long position) {
      * @param bytes The byte array containing the serialized deletion vector.
      * @return A DeletionVector instance that represents the deserialized data.
      */
-    static DeletionVector deserializeFromBytes(byte[] bytes) {
-        try {
-            ByteBuffer buffer = ByteBuffer.wrap(bytes);
-            int magicNum = buffer.getInt();
-            if (magicNum == BitmapDeletionVector.MAGIC_NUMBER) {
-                return BitmapDeletionVector.deserializeFromByteBuffer(buffer);
-            } else {
-                throw new RuntimeException("Invalid magic number: " + magicNum);
-            }
-        } catch (IOException e) {
-            throw new RuntimeException("Unable to deserialize deletion vector", e);
+    static DeletionVector deserializeFromBytes(byte[] bytes, int version) {
+        if (version == BitmapDeletionVector.VERSION) {
+            return BitmapDeletionVector.deserializeFromBytes(bytes);
+        } else if (version == Bitmap64DeletionVector.VERSION) {
+            return Bitmap64DeletionVector.deserializeFromBytes(bytes);
+        } else {
+            throw new RuntimeException("Invalid deletion vector version: " + version);
         }
     }
 
     static DeletionVector read(FileIO fileIO, DeletionFile deletionFile) throws IOException {
         Path path = new Path(deletionFile.path());
         try (SeekableInputStream input = fileIO.newInputStream(path)) {
+            // read dv version
+            int version = input.read();
```

Comment:
Can we avoid this reading? Can we just use magic number to decide which version?